src/components/Pool/PairView/index.tsx

```diff
@@ -0,0 +1,112 @@
+import { usePair24hVolumeUSD } from '../../../hooks/usePairVolume24hUSD'
+import { usePairLiquidityUSD } from '../../../hooks/usePairLiquidityUSD'
+import LiquidityMiningCampaigns from './LiquidityMiningCampaigns'
+import { useActiveWeb3React } from '../../../hooks'
+
+const StyledDarkCard = styled(DarkCard)`
+  ::before {
+    background: ${props => props.theme.bg1};
+  }
+`
+
+const DataText = styled.div`
+  font-size: 14px;
+  line-height: 17px;
+  font-weight: 500;
+  color: ${props => props.theme.purple2};
+`
+
+interface DataRowProps {
+  loading?: boolean
+  title: string
+  value: ReactNode
+}
+
+function DataRow({ title, value, loading }: DataRowProps) {
+  return (
+    <RowBetween justify="space-between" width="100%" marginBottom="8px">
+      <DataText>{title}</DataText>
+      <DataText>{loading ? <Skeleton width="36px" /> : value}</DataText>
+    </RowBetween>
+  )
+}
+
+interface PairViewProps {
+  loading: boolean
+  pair?: Pair | null
+}
+
+function PairView({ loading, pair }: PairViewProps) {
+  const { account } = useActiveWeb3React()
+  const { loading: volumeLoading, volume24hUSD } = usePair24hVolumeUSD(pair)
+  const { loading: liquidityLoading, liquidityUSD } = usePairLiquidityUSD(pair)
+  const liquidityMiningEnabled = useLiquidityMiningFeatureFlag()
+
+  const overallLoading = loading || volumeLoading || liquidityLoading
+
+  return (
+    <>
+      <StyledDarkCard padding="32px">
+        <Flex flexDirection="column">
+          <Flex mb="18px" alignItems="center">
+            <Box mr="8px">
+              <DoubleCurrencyLogo
+                loading={overallLoading}
+                size={20}
+                currency0={pair?.token0}
+                currency1={pair?.token1}
+              />
+            </Box>
+            <Box>
+              <Text fontSize="16px" fontWeight="600" lineHeight="20px">
+                {pair ? `${pair?.token0.symbol}/${pair?.token1.symbol}` : <Skeleton width="60px" />}
+              </Text>
+            </Box>
+          </Flex>
+          <DataRow loading={overallLoading} title="Liquidity:" value={`$${liquidityUSD.toFixed(2)}`} />
+          <DataRow loading={overallLoading} title="Volume:" value={`$${volume24hUSD.toFixed(2)}`} />
+          {!!account && (
+            <Box mt="18px">
+              <FullPositionCard pair={pair || undefined} />
+            </Box>
+          )}
+          <RowBetween mt="18px">
+            <ButtonGrey
+              padding="8px"
+              disabled
+              style={{ fontSize: '12px', fontWeight: 'bold', lineHeight: '15px' }}
+              width="100%"
+            >
+              GOVERNANCE
+            </ButtonGrey>
+          </RowBetween>
+        </Flex>
+      </StyledDarkCard>
+      {liquidityMiningEnabled && <LiquidityMiningCampaigns pair={pair || undefined} />}
+    </>
+  )
+}
+
+export default memo(PairView, (previousProps, nextProps) => {
+  // avoids pair reference changes to mess things up by reloading the whole thing
```

Comment:
Not sure I understand "avoids pair reference changes to mess things up"

Reply:
Basically, if the exact same pair got passed in, but with a different memory reference (same pair, different object instance), React would see this as a change in values that potentially impacts the component's logic, triggering a re-render in the process. With this explicit `memo` function, we manually check if the 2 pairs (the "current" one and the "next" one) are equivalent, and even if they have a different memory reference, we avoid re-rendering.